nats/src/main/java/io/micronaut/nats/connect/NatsConnectionFactoryConfig.java

```diff
@@ -424,4 +464,101 @@ private SSLContext createTlsContext() throws IOException, GeneralSecurityExcepti
         }
 
     }
+
+    /**
+     * Manages the jetstream configuration.
+     *
+     * @author Joachim Grimm
+     * @since 4.0.0
+     */
+    @Introspected
+    @ConfigurationProperties("jetstream")
+    public static class JetStreamConfiguration {
+
+        @ConfigurationBuilder(prefixes = "")
+        private JetStreamOptions.Builder builder =
+            JetStreamOptions.builder(JetStreamOptions.defaultOptions());
+
+        private Map<String, StreamConfiguration> streams = new HashMap<>();
+
+        /**
+         * get the jetstream options builder.
+         *
+         * @return options builder
+         */
+        public JetStreamOptions.Builder getBuilder() {
+            return builder;
+        }
+
+        /**
+         * return the configuration as {@link JetStreamOptions}.
+         *
+         * @return jetstream options
+         */
+        public JetStreamOptions toJetStreamOptions() {
+            return builder.build();
+        }
+
+        /**
+         * get the stream configurations.
+         *
+         * @return list of streamConfigurations
+         */
+        public Map<String, StreamConfiguration> getStreams() {
+            return streams;
+        }
+
+        /**
+         * set the stream configurations for the jetstream.
+         *
+         * @param streams the stream configurations
+         */
+        public void setStreams(Map<String, StreamConfiguration> streams) {
```

Comment:
Does this actually work? You may need to `@Inject` on this method with `List<StreamConfiguration>` and in `StreamConfiguration` add a constructor like:

```java
StreamConfiguration(@Parameter String name) {

}
```

You can then populate the map from the names

Reply:
It works perfectly. I used this approach, because the injection is not working on nested `@EachProperty`. This is this issue in mn-core: micronaut-projects/micronaut-core#3938

Reply:
this should be fixed in 4.0.x